### PR TITLE
Dockerignore

### PR DESCRIPTION
**Pull request by wyaeld**
Added 2 new integration-cli tests for the dockerignore feature

### CODE REVIEW

**Comment by wyaeld** [comment]
I'm having a minor problem with the 2nd test that has stumped me.

The test fails with

```
--- FAIL: TestDockerignoreWithMultiplePaths (0.01 seconds)
    utils.go:123: build failed to complete: Uploading context  2.56 kB
        Uploading context 
        2014/06/01 08:58:45 Error: Can't build a directory with no Dockerfile
         exit status 1
```

I cannot work out why its complaining that it can't see the Dockerfile in that dir.

File: integration-cli/build_tests/TestDockerignore/IgnoreFiltersMultiplePaths/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# testfile is dockerignored
+RUN [ -z $(ls -l | grep test_file | awk '{print $9}') ]
+
+# test_file2 is not
+RUN [ -f $(ls -l | grep test_file2 | awk '{print $9}') ]
+
+# dotfiles and dirs can also be hidden
+RUN [ -z $(ls -l | grep .hidden_file | awk '{print $9}') ]
+RUN [ -z $(ls -l | grep .hidden_dir | awk '{print $9}') ]
```
Comment (LK4D4) [inline integration-cli/build_tests/TestDockerignore/IgnoreFiltersMultiplePaths/Dockerfile]: You can try file existence just with [ -e file ]

**Comment by wyaeld** [comment]
updated, my sh skills need work.

**Comment by wyaeld** [comment]
Added another test, 2/3 are still failing though for reasons I can't figure out.

**Comment by wyaeld** [comment]
now rebased on dotcloud/docker

**Comment by wyaeld** [comment]
I managed to screw this up with the rebase, going to close and do it again cleanly

**Comment by tmc** [comment]
Cool, no problem

**Comment by wyaeld** [comment]
This time its the correct commits, without any mistakes in the rebase.

Still 2 problems.

`TestDockerignoreCanIgnoreDockerfile` fails, but looking closer at the interaction of the builder, it actually builds using the tar that it receives from archive?  That would mean dockerignoring the actual Dockerfile is never going to work.

`TestDockerignoreWithMultiplePaths` fails too, but this one I can't understand.  The dockerignore doesn't mention the Dockerfile at all, but the builder still complains that it didn't get a Dockerfile in the archive.

**Comment by LK4D4** [comment]
@wyaeld Yeah, ignoring Dockerfile doesn't make sense, I think we should put check about it in `api`.
I don't see also what the problem with multiple paths, but if you have time you can put some debug messages to `archive` and see what's really going on. Feel free ask me for help.

**Comment by wyaeld** [comment]
Ok, so the `TestDockerignoreCanIgnoreDockerfile` isn't valid, and needs to be removed and the behaviour documented, plus probably a check as you say.

`TestDockerignoreWithMultiplePaths` still a problem.  I haven't had much time yet, will have some in a couple of days, but any assistance appreciated

**Pull request closed without merging**